docs/topics/media-pipeline.rst

```diff
@@ -258,11 +258,21 @@ For self-hosting you also might feel the need not to use SSL and not to verify S
 
     AWS_USE_SSL = False # or True (None by default)
     AWS_VERIFY = False # or True (None by default)
-
-.. _botocore: https://github.com/boto/botocore
-.. _canned ACLs: https://docs.aws.amazon.com/AmazonS3/latest/dev/acl-overview.html#canned-acl
+    
 .. _Minio: https://github.com/minio/minio
 .. _s3.scality: https://s3.scality.com/
+.. _botocore: https://github.com/boto/botocore
+.. _canned ACLs: https://docs.aws.amazon.com/AmazonS3/latest/dev/acl-overview.html#canned-acl
+
+Custom Amazon S3 headers
+''''''''''''''''''''''''
+
+To send custom headers to AWS S3 when using :class:`~scrapy.pipelines.files.FilesPipeline`
+or :class:`~scrapy.pipelines.images.ImagesPipeline`, follow these steps:
+
+Subclass :class:`~scrapy.pipelines.files.S3FilesStore`, Extend its ``HEADERS`` class attribute in your subclass to define the headers you want with the values you want.
+For reference, `S3 <https://botocore.amazonaws.com/v1/documentation/api/latest/reference/services/s3.html>`_ , `Common Request Headers <https://docs.aws.amazon.com/AmazonS3/latest/API/RESTCommonRequestHeaders.html>`_ , `Common Response Headers <https://docs.aws.amazon.com/AmazonS3/latest/API/RESTCommonResponseHeaders.html>`_ , `Request Response Data Mapping <https://docs.aws.amazon.com/apigateway/latest/developerguide/request-response-data-mappings.html>`_.
+You can see the header-to-key mapping in the source code of the class, for additional header names then, Subclass ``FilesPipeline``, and edit the ``STORE_SCHEMES`` class attribute in your subclass to point ``s3`` to your ``S3FilesStore`` subclass. Update your ``ITEM_PIPELINES`` setting to use your ``FilesPipeline`` subclass.
```

Comment:
```suggestion
.. _botocore: https://github.com/boto/botocore
.. _canned ACLs: https://docs.aws.amazon.com/AmazonS3/latest/dev/acl-overview.html#canned-acl
.. _Minio: https://github.com/minio/minio
.. _s3.scality: https://s3.scality.com/


Custom Amazon S3 headers
'''''''''''''''''''''''''

To send custom headers to AWS S3 when using :class:`~scrapy.pipelines.files.FilesPipeline`
or :class:`~scrapy.pipelines.images.ImagesPipeline`, follow these steps:

#. Subclass :class:`~scrapy.pipelines.files.S3FilesStore`, 

#. Extend its ``HEADERS`` class attribute in your subclass to define the headers you want with the values you want. For reference,  `S3 <https://botocore.amazonaws.com/v1/documentation/api/latest/reference/services/s3.html>`_ , `Common Request Headers <https://docs.aws.amazon.com/AmazonS3/latest/API/RESTCommonRequestHeaders.html>`_ , `Common Response Headers <https://docs.aws.amazon.com/AmazonS3/latest/API/RESTCommonResponseHeaders.html>`_ , `Request Response Data Mapping <https://docs.aws.amazon.com/apigateway/latest/developerguide/request-response-data-mappings.html>`_

#. You can see the header-to-key mapping in the source code of the class, for additional header names,

#. Subclass ``FilesPipeline``, and edit the ``STORE_SCHEMES`` class attribute in your subclass to point ``s3`` to your ``S3FilesStore`` subclass,

#. Update your ``ITEM_PIPELINES`` setting to use your ``FilesPipeline`` subclass.
```

Reply:
I think step 3 should be a second paragraph of step 2, rather than a step.